elephant/test/test_spike_train_generation.py

```diff
@@ -69,6 +68,66 @@ def test_threshold_detection(self):
         except AttributeError: # If numpy version too old to have allclose
             self.assertTrue(np.array_equal(spike_train,spike_train))
 
+
+class AnalogSignalPeakDetectionTestCase(unittest.TestCase):
+
+    def setUp(self):
+        curr_dir = os.path.dirname(os.path.realpath(__file__))
+        npz_file_loc = os.path.join(curr_dir, 'spike_extraction_test_data.npz')
+        iom2 = neo.io.PyNNNumpyIO(npz_file_loc)
+        data = iom2.read()
+        self.vm = data[0].segments[0].analogsignals[0]
+        self.true_time_stamps = [0.0124,  0.0354,  0.0713,  0.1192,  0.1695,
+                                 0.2201,  0.2711] * second
+
+    def test_peak_detection_time_stamps(self):
+        # Test with default arguments
+        result = stgen.peak_detection(self.vm)
+        self.assertEqual(len(self.true_time_stamps), len(result))
+        self.assertIsInstance(result, neo.core.SpikeTrain)
+
+        try:
```

Comment:
If the integers are same then `assert_array_almost_equal` is anyway not giving an error. So, I think you could just leave only the `assert_array_almost_equal` without the try except block. Also In the  `AnalogSignalThresholdDetectionTestCase` I am wondering what it does. Furthermore, it excepts an `AttributeError` whereas  I would expect an `AssertionError`.